iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveTableUtil.java

```diff
@@ -0,0 +1,142 @@
+      try {
+        return !name.startsWith("_") && !name.startsWith(".") && fileSystem.isFile(path);
+      } catch (IOException e) {
+        return false;
+      }
+    }
+  }
+
+  private HiveTableUtil() {
+  }
+
+  /**
+   * Import files from given partitions to an Iceberg table.
+   * @param sourceLocation location of the HMS table
+   * @param format inputformat class name of the HMS table
+   * @param partitionSpecProxy  list of HMS table partitions wrapped in partitionSpecProxy
+   * @param icebergTable destination iceberg table
+   * @param conf a Hadoop configuration
+   */
+  public static void importFiles(String sourceLocation, String format, PartitionSpecProxy partitionSpecProxy,
+      Table icebergTable, Configuration conf) {
+    AppendFiles append = icebergTable.newAppend();
+    try {
+      URI uri = new URI(sourceLocation);
+      FileSystem fileSystem = FileSystem.get(uri, conf);
+      PartitionSpec spec = icebergTable.spec();
+      MetricsConfig metricsConfig = MetricsConfig.fromProperties(icebergTable.properties());
+      String nameMappingString = icebergTable.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+      NameMapping nameMapping = nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+      if (partitionSpecProxy.size() == 0) {
+        List<DataFile> dataFiles = getDataFiles(fileSystem, new Path(uri), Collections.emptyMap(), format, spec,
+            metricsConfig, nameMapping, conf);
+        dataFiles.forEach(append::appendFile);
+      } else {
+        PartitionSpecProxy.PartitionIterator iterator = partitionSpecProxy.getPartitionIterator();
+        List<Callable<List<DataFile>>> tasks = new ArrayList<>();
+        while (iterator.hasNext()) {
+          Partition partition = iterator.next();
+          Callable<List<DataFile>> task = () -> {
+            Path partitionPath = new Path(partition.getSd().getLocation());
+            Map<String, String> partitionSpec = Warehouse.makeSpecFromName(partitionPath.toString());
+            return getDataFiles(fileSystem, partitionPath, partitionSpec, format.toLowerCase(), spec, metricsConfig,
+                nameMapping, conf);
+          };
+          tasks.add(task);
+        }
+        int numThreads = HiveConf.getIntVar(conf, HiveConf.ConfVars.HIVE_SERVER2_ICEBERG_METADATA_GENERATOR_THREADS);
+        ExecutorService executor = Executors.newFixedThreadPool(numThreads,
+            new ThreadFactoryBuilder().setNameFormat("Iceberg metadata generator").setDaemon(true).build());
+        List<Future<List<DataFile>>> futures = executor.invokeAll(tasks);
+        for (Future<List<DataFile>> future : futures) {
+          List<DataFile> dataFiles = future.get();
+          dataFiles.forEach(append::appendFile);
+        }
+      }
+
+    } catch (URISyntaxException | IOException | InterruptedException | ExecutionException e) {
+      LOG.error("Cannot import hive data into iceberg table", e);
```

Comment:
Are we sure want to swallow the exception here? If the import failed or only partially succeeded, we're still committing the appends below and not fail the operation. Couldn't this lead to an inconsistent state and silently hiding failures from the user?

Reply:
You are correct. I refactored the code to get back with an exception.